x-pack/test/security_solution_endpoint/apps/endpoint/endpoint_list.ts

```diff
@@ -94,137 +96,176 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
-          await adminSearchBar.type('na');
-          const querySubmitButton = await testSubjects.find('querySubmitButton');
-          await querySubmitButton.click();
-          const expectedDataFromQuery = [
-            [
-              'Endpoint',
-              'Agent status',
-              'Policy',
-              'Policy status',
-              'OS',
-              'IP address',
-              'Version',
-              'Last active',
-              'Actions',
-            ],
-            ['No items found'],
-          ];
-
-          await pageObjects.endpoint.waitForTableToNotHaveData('endpointListTable', 10000);
-          const tableData = await pageObjects.endpointPageUtils.tableData('endpointListTable');
-          expect(tableData).to.eql(expectedDataFromQuery);
+
+        it('finds page title', async () => {
+          const title = await testSubjects.getVisibleText('header-page-title');
+          expect(title).to.equal('Endpoints');
         });
 
-        it('when the kql filters for united.endpoint.host.hostname, table shows 1 item', async () => {
-          const expectedDataFromQuery = [...expectedData.slice(0, 2).map((row) => [...row])];
-          const hostName = expectedDataFromQuery[1][0];
-          const adminSearchBar = await testSubjects.find('adminSearchBar');
-          await adminSearchBar.clearValueWithKeyboard();
-          await adminSearchBar.type(
-            `united.endpoint.host.hostname : "${hostName}" or host.hostname : "${hostName}" `
-          );
-          const querySubmitButton = await testSubjects.find('querySubmitButton');
-          await querySubmitButton.click();
-          await pageObjects.endpoint.waitForTableToHaveNumberOfEntries(
-            'endpointListTable',
-            1,
-            90000
-          );
+        it('displays table data', async () => {
           const tableData = await formattedTableData();
-          expect(tableData.sort()).to.eql(expectedDataFromQuery.sort());
+          expect(tableData.sort()).to.eql(expectedData.sort());
         });
-      });
-      it('does not show the details flyout initially', async () => {
-        await testSubjects.missingOrFail('endpointDetailsFlyout');
-      });
 
-      describe('when the hostname is clicked on,', () => {
-        before(async () => {
-          await pageObjects.endpoint.waitForTableToHaveNumberOfEntries(
-            'endpointListTable',
-            3,
-            90000
-          );
+        describe('for the search bar', () => {
+          before(async () => {
+            await pageObjects.endpoint.waitForTableToHaveData('endpointListTable', 60000);
+          });
+          after(async () => {
+            const adminSearchBar = await testSubjects.find('adminSearchBar');
+            const querySubmitButton = await testSubjects.find('querySubmitButton');
+            await adminSearchBar.clearValueWithKeyboard();
+            await querySubmitButton.click();
```

Comment:
FYI: We have had trouble in the back with "stale elements" when we first query for a Button and then later `.click()` on it. The best practice is to use `testSubjects.click(testSubj)` instead. Same comment for other areas in this PR where you are doing this.